control-plane/agents/src/bin/core/controller/registry.rs

```diff
@@ -130,19 +131,25 @@ impl Registry {
                 reconcile_period,
                 reconcile_idle_period,
                 reconciler: ReconcilerControl::new(),
-                config: Self::get_config_or_panic(&mut store, legacy_prefix_present).await,
+                config: parking_lot::RwLock::new(
+                    Self::get_config_or_panic(&mut store, legacy_prefix_present).await,
+                ),
                 max_rebuilds,
                 host_acl,
                 legacy_prefix_present,
             }),
         };
         registry.init().await;
 
-        // Remove all entries of v1 key prefix.
-        store
-            .delete_values_prefix(&product_v1_key_prefix())
-            .await
-            .expect("Product v1 key prefix deletion should not fail");
+        // Disable v1 compat if nexus_info keys are migrated.
+        if !registry.nexus_info_v1_present().await {
```

Comment:
hmm shouldn't this be negated, if v1 is present?
Also I think this should pre preceded by if compat is enable to avoid doing this everytime?
registry.compat_v1() && registry.nexus_info_v1_present()

Reply:
The method returns true if the some v1 nexus_info is not migrated so we are cleaning up only when the method returns false, i.e. all keys migrated.

The v1 compat enable check is already a part of the method. We don't need to check it here.

Reply:
Yeah probably the method name is wrong here, any suggestions?

Reply:
hmm the method doesn't check if compat is enabled at the start AFAICT... only if it failed to load the nexus info.. 
if compat is not enabled, then we don't need to do anything right? which is why I suggested we check if it's enabled here, though we can also check at the start of that function

Reply:
Ahh okay. You mean don't do anything at all. Cool. I will move that check up. Thanks